File: uniapp-frontend/src/model/Student.ts
import { Attendance } from "./Attendance";
import { Enrollment } from "./Enrollment";
import { Grade } from "./Grade";
import { Role } from "./Role";
import { User } from "./User";

export interface Student extends User {
    attendances : Attendance[]
    grades : Grade[]
    enrollments : Enrollment[]
    role : Role.STUDENT
}

File: uniapp-server/src/main/java/dev/godraadam/uniapp/model/Student.java
package dev.godraadam.uniapp.model;

import java.util.List;

import javax.persistence.Entity;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "students")
@Getter
@Setter
public class Student extends ApplicationUser {
    
    private String groupName;

    @OneToMany(mappedBy = "student")
    private List<Attendance> attendances;

    @OneToMany(mappedBy = "student")
    private List<Grade> grades;

    @OneToMany(mappedBy = "student")
    private List<Enrollment> enrollments;

    private Role role = Role.STUDENT;
}
